docx/text/paragraph.py

```diff
@@ -62,6 +62,36 @@ def clear(self):
         self._p.clear_content()
         return self
 
+    @property
+    def footnotes(self):
+        """
+        Returns a list of |Footnote| instances that refers to the footnotes in this paragraph,
+        or |None| if none footnote is defined.
+        """
+        reference_ids = self.footnote_reference_ids
+        if reference_ids == None:
+            return None
+        footnotes = self._parent._parent.footnotes
+        footnote_list = []
+        for ref_id in reference_ids:
+            footnote_list.append(footnotes[ref_id])
+        return footnote_list
+
+    @property
+    def footnote_reference_ids(self) -> (list[int]|None):
+        """
+        Return all footnote reference ids (``<w:footnoteReference>``) form the paragraph,
+        or |None| if not present.
+        """
+        footnote_ids = []
+        for run in self.runs:
+            new_footnote_ids = run.footnote_reference_ids
+            if new_footnote_ids:
+                footnote_ids.extend(new_footnote_ids)
+        if footnote_ids == []:
+            footnote_ids = None
```

Comment:
Would it make sense to return an empty list if there are no footnotes? I can see the benefit that users don't have to deal with `None` but instead check if the list is empty. What are your thoughts on this?

Reply:
I saw that for paragraph and run properties the library returns None, but I agree, we can return an empty array.